### PR TITLE
Enhance testdisk

### LINKED ISSUE #3879: SynoCli Disk Tools testdisk missing e2fsprogs option
testdisk appears to not be compiled with e2fsprogs. This was noticed when accessing a ext4 partition via testdisk on Windows/Ubuntu but gives a compilation message when accessing via DSM. I have tried compiling the package with some settings added but have not been successful. Requesting that e2fsprogs be added to the build or some advice on compiling it.

### Setup
_Package Name:_ SynoCli Disk Tools
_Package Version:_ 1.0-1

_NAS Model:_ DS218+
_NAS Architecture:_ apollolake
_DSM version:_ 6.2.2-24922 Update 4 (package compiled as 6.1)

### Expected behavior
testdisk should be able to access a ext4 partition

### Actual behavior
testdisk is not able to access a ext4 partition

### Steps to reproduce
_1._ `:$ sudo testdisk -v`
_2._ notice ext2fs lib: none
_3._ edit makefile for cross/testdisk ([https://www.cgsecurity.org/wiki/TestDisk_Compilation](https://www.cgsecurity.org/wiki/TestDisk_Compilation)):
`DEPENDS = cross/ncursesw cross/libuuid cross/e2fsprogs`
`DEP_E2FSPROGS = $(WORK_DIR)/e2fsprogs-1.45.4`
`DEP_E2FSPROGS_LIB = $(DEP_E2FSPROGS)/lib/`
`DEP_E2FSPROGS_INCLUDE = $(DEP_E2FSPROGS)/include/`
`CONFIGURE_ARGS = --with-ext2fs-lib=$(DEP_E2FSPROGS_LIB) --with-ext2fs-includes=$(DEP_E2FSPROGS_INCLUDE)`
_4._ uninstall, compile, and install new spk
_5._ `:$ sudo testdisk -v`
_6._ still ext2fs lib: none

#### Package log
_Check Package Center or `/usr/local/{package}/var/`_
```
NA
```
#### Other logs
_E.g. `/var/log/messages` or `/var/log/synopkg.log`_
```
NA
```

### PR DESCRIPTION
**Pull request by hgy59**
_Motivation:_  Some features of TestDisk are missing
_Linked issues:_  #3879

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### Remarks
- add ntfs, reiserfs, ewf and jpeg support
- add e2fsprogs (closes #3879)
- add ntfsprogs

### CODE REVIEW

**Comment by hgy59** [comment]
Added support for all file systems where cross module build succeeded.
But don't know how to verify reiser and ewf formats.

**Comment by hgy59** [comment]
I have published a pre-release of this in my [fork of spksrc](https://github.com/hgy59/spksrc/releases/tag/testdisk_7.1_4_pre)

@vizbird can you test whether e2fsprogs works for you?

As usual this is a prerelease: _without any warranty_

**Comment by vizbird** [comment]
Yep. I'll test out this week.

**Comment by ymartin59** [comment]
@hgy59 To test other file systems, it may be possible to create a dedicated volume on DSM with that concerned file system before running testdisk on it... Or else create a disk image file thanks to Linux loopback device, and testdisk is expected to be able to read it too.

**Comment by vizbird** [comment]
SynoCLI-Disk 3.0-3 was compiled for apollolake 6.1. I'm getting a "Failed To Install" error message when installing via dsm package manager. I compiled 2.0-2 from master just to test and was able to get it to install. My DSM is on 6.2 so not sure if that may be an issue. I saved the compilation log if there is something in there that can help.

**Comment by hgy59** [comment]
> 
> 
> SynoCLI-Disk 3.0-3 was compiled for apollolake 6.1. I'm getting a "Failed To Install" error message when installing via dsm package manager. I compiled 2.0-2 from master just to test and was able to get it to install. My DSM is on 6.2 so not sure if that may be an issue. I saved the compilation log if there is something in there that can help.

Installation is fixed now. Packages created by github-action can be downloaded [(here)](https://github.com/SynoCommunity/spksrc/actions/runs/158385887)

File: cross/testdisk.reiserfs/Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+PKG_NAME = progsreiserfs
```
Comment (ymartin59) [inline cross/testdisk.reiserfs/Makefile:1]: Please rename folder as `cross/progsreiserfs` for homogeneity

**Pull request merged**